arch/arm/soc/atmel_sam0/samd20/soc.c

```diff
@@ -0,0 +1,196 @@
+/*
```

Comment:
WDYT about making a atmel_sam0/common directory and putting this in there instead?  This is how the atmel_sam/ common code is handled.

Reply:
They are not 100% identical, if we choose to add a common directory, then we would have to deal with some if defs